core/bcast/bcast.go

```diff
@@ -299,3 +307,27 @@ func newDelayFunc(ctx context.Context, eth2Cl eth2wrap.Client) (func(slot uint64
 		return time.Since(slotStart)
 	}, nil
 }
+
+// firstSlotInCurrentEpoch calculates first slot number of the current ongoing epoch.
+func firstSlotInCurrentEpoch(ctx context.Context, eth2Cl eth2wrap.Client) (uint64, error) {
+	genesis, err := eth2Cl.Genesis(ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "fetch genesis")
+	}
+
+	slotDuration, err := eth2Cl.SlotDuration(ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "fetch slot duration")
+	}
+
+	slotsPerEpoch, err := eth2Cl.SlotsPerEpoch(ctx)
```

Comment:
This is deprecated too, use `Spec` instead

Reply:
Ignoring this for now